compiler/include/ForallStmt.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _FORALL_STMT_H_
+#define _FORALL_STMT_H_
+
+#include "stmt.h"
+
+// forall loop statement
+
+class ForallStmt : public Stmt
+{
+public:
+  bool       zippered()       const; // was 'zip' keyword used?
+  AList&     inductionVariables();   // DefExprs, one per iterated expr
+  AList&     iteratedExpressions();  // SymExprs, CallExprs
+  AList&     intentVariables();      // DefExprs of LoopIntentVars with intents
```

Comment:
Also, a nit, but is there a reason not to make inductionVariables, iteratedExpressions, and intentVariables have const this and return a const reference?